### ISSUE TITLE
MissingMethodException using Ravendb.Client and NodaTime.Serialization.JsonNet

### ISSUE THREAD

**Issue by germanftorres** [filipw/dotnet-script#214]
Hi!

I get an exception while referencing various packages that depend on Json.NET (RavenDB.Client and NodaTime.Serialization.JsonNET.  I guess it's related to assembly binding redidrect.  Creating a dotnet new console application with the exact same packages runs fine.

Any ideas what's going on?

Thanks!!

Germán

```csharp
#! "netcoreapp2.0"
#r "nuget: NodaTime, 2.0.0"
#r "nuget: NodaTime.Serialization.JsonNet, 2.0.0"
#r "nuget: RavenDB.Client, 4.0.0-rc-40025"

using System;
using System.Collections.Generic;
using System.Linq;
using NodaTime;
using NodaTime.Serialization.JsonNet;

var converter = NodaConverters.InstantConverter;
Console.WriteLine("Hello World!");

```
```
System.MissingMethodException: Method not found: 'Newtonsoft.Json.JsonConverter NodaTime.Serialization.JsonNet.NodaConverters.get_InstantConverter()'.
   at Submission#0.<<Initialize>>d__0.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine)
   at Submission#0.<Initialize>()
   at Submission#0.<Factory>(Object[] submissionArray)
   at Microsoft.CodeAnalysis.Scripting.ScriptExecutionState.<RunSubmissionsAsync>d__9`1.MoveNext()

```

**Comment by seesharper** [comment]
Hi and thanks for reporting this. We have confirmed this to be a bug and will be looking into it. 

What we have found so far is that RavenDb pulls in Newtonsoft.JSon 10.0.3 which is the version that we add as a reference to the script. The same version (10.0.3) is also loaded if we create a standard console app with these dependencies. The examples here are simplified to pull in Newtonsoft.Json 10.0.3 directly instead of via RavenDB: The error is still present. 

ConsoleApp.csproj
```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp2.0</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="NodaTime" Version="2.2.3" />
    <PackageReference Include="NodaTime.Serialization.JsonNet" Version="2.0.0" />
    <PackageReference Include="Newtonsoft.Json" Version="10.0.3" />  
  </ItemGroup>

</Project>
```

Script.csproj
```
<?xml version="1.0" encoding="utf-8"?>                                           
<Project Sdk="Microsoft.NET.Sdk">                                                
  <PropertyGroup>                                                                
    <OutputType>Exe</OutputType>                                                 
    <TargetFramework>netcoreapp2.0</TargetFramework>                             
  </PropertyGroup>                                                               
  <ItemGroup>                                                                    
    <PackageReference Include="NodaTime" Version="2.0.0" />                      
    <PackageReference Include="NodaTime.Serialization.JsonNet" Version="2.0.0" />
    <PackageReference Include="Newtonsoft.Json" Version="10.0.3" />              
  </ItemGroup>                                                                   
</Project>                                                                       
```
The big question here is why we get a `MethodNotFoundException` since we load the same assemblies both in the script and in the console app. 

Also tried to recompile `NodaTime.Serialization` with `Newtonsoft.Json 10.0.3` just to see if it compiles without modification. No problems there.

**Comment by filipw** [comment]
fixed by #215
